src/extensions/snapshot/hooks.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+const path = require('path')
+const _ = require('lodash')
+
+const clean = require('./clean')
+
+/**
+ * Registers hooks for the fixtures extension.
+ *
+ * @module extensions/fixtures/hooks
+ */
+
+module.exports = ({ registerHandler, Before, BeforeAll, AfterAll }) => {
+    Before(function(scenarioInfos) {
+        let file = null
+        let line = null
+
+        if (scenarioInfos.sourceLocation) {
+            file = scenarioInfos.sourceLocation.uri
+            line = scenarioInfos.sourceLocation.line
+        } else {
+            const fullPath = scenarioInfos.scenario.feature.uri
+            const relativePath = path.relative(process.cwd(), fullPath)
+            file = relativePath
+            line = scenarioInfos.scenario.line
+        }
+
+        this.snapshot.featureFile = file
+        this.snapshot.scenarioLine = line
+    })
+
+    if (registerHandler) {
+        registerHandler('BeforeFeatures', function() {
+            clean.resetReferences()
+        })
+
+        registerHandler('AfterFeatures', function() {
+            if (!_.isEmpty(_.intersection(process.argv, ['--cleanSnapshots']))) {
```

Comment:
You already computed this in `src/extensions/snapshot/extend_world.js`, cannot we retrieve this from the extension itself?

Reply:
what do you think about using veggies itself? You could run it from a separate directory for example and check generated snapshots using file extension or even the snapshot one (a little bit of inception I admit), it could work.